meson.build

```
@@ -0,0 +1,502 @@
+  ]
+  qjs_native_lib = static_library(
+    'qjs_native',
+    qjs_srcs,
+    qjs_libc_srcs,
+
+    dependencies: qjs_sys_native_deps,
+    c_args: qjs_c_args,
+    gnu_symbol_visibility: 'hidden',
+
+    build_by_default: false,
+    native: true,
+    install: false,
+  )
+
+  meson.override_find_program(
+    'qjsc',
+    executable(
+      'qjsc_native',
+      qjsc_srcs,
+
+      c_args: qjs_c_args,
+      link_with: qjs_native_lib,
+      dependencies: qjs_sys_native_deps,
+
+      build_by_default: false,
+      native: true,
+      install: false,
+    ),
+  )
+  meson.override_find_program(
+    'qjs',
+    executable(
+      'qjs_native',
+      qjs_exe_srcs,
+
+      c_args: qjs_c_args,
+      link_with: qjs_native_lib,
+      dependencies: [qjs_sys_native_deps, mimalloc_native_dep],
+
+      build_by_default: false,
+      native: true,
+      install: false,
+    ),
+  )
+else
+  meson.override_find_program('qjsc', qjsc_exe)
+  meson.override_find_program('qjs', qjs_exe)
+endif
+
+tests = get_option('tests').disable_auto_if(meson.is_subproject())
+examples = get_option('examples').disable_auto_if(meson.is_subproject())
+
+if tests.allowed()
+  if host_system != 'emscripten'
+    # Test262 runner
+    run262_exe = executable(
+      'run-test262',
+      'run-test262.c',
+      qjs_libc_srcs,
+
+      c_args: qjs_c_args,
+      dependencies: qjs_dep,
+    )
+
+    test(
+      'test',
+      run262_exe,
+      args: ['-c', files('tests.conf')],
+      workdir: meson.current_source_dir(),
+    )
+
+    foreach bench : [
+      'empty_loop',
+
+      'date_now',
+
+      'prop_read',
+      'prop_write',
+      'prop_create',
+      'prop_delete',
+
+      'array_read',
+      'array_write',
+      'array_prop_create',
+      'array_length_decr',
+      'array_hole_length_decr',
+      'array_push',
+      'array_pop',
+
+      'typed_array_read',
+      'typed_array_write',
+
+      'global_read',
+      'global_write',
+      'global_write_strict',
+
+      'local_destruct',
+      'global_destruct',
+      'global_destruct_strict',
+
+      'func_call',
+      'closure_var',
+
+      'int_arith',
+      'float_arith',
+
+      'set_collection_add',
+
+      'array_for',
+      'array_for_in',
+      'array_for_of',
+
+      'math_min',
+
+      'regexp_ascii',
+      'regexp_utf16',
+
+      'string_build1',
+      'string_build2',
+
+      'sort_bench',
+
+      'int_to_string',
+      'int_toString',
+
+      'float_to_string',
+      'float_toString',
+      'float_toFixed',
+      'float_toPrecision',
+      'float_toExponential',
+
+      'string_to_int',
+      'string_to_float',
+
+      'bigint64_arith',
+      'bigint256_arith',
+    ]
+      benchmark(
+        bench,
+        qjs_exe,
+        args: [files('tests/microbench.js'), bench],
+      )
+    endforeach
+  endif
+
+  # Interrupt test
+  test(
+    'interrupt',
+    executable(
+      'interrupt-test',
+      'interrupt-test.c',
```

Was renamed in 22cd6ab711fffc471d7fde99536fb6bbb033184b.
```suggestion
  # API test
  test(
    'api',
    executable(
      'api-test',
      'api-test.c',
```

---

done